File: Assignment Programs/Assignment18/program5.c
#include<stdio.h>
#include<stdlib.h>

int Product(int Arr[], int iLength)
{
	int iCnt = 0, iMult = 1, iFlag = 0;
	for(iCnt = 0;iCnt<iLength;iCnt++)
	{
		if((Arr[iCnt] %2 != 0))
		{
			iFlag++;
			iMult = iMult * Arr[iCnt];
		}
	}
	if(iFlag != 0)
	{
		return iMult;
	}
	else
	{
		return 0;
	}
	
}

int main()
{
	int iSize = 0, iRet = 0, iCnt = 0;
	int *p = NULL;
	printf("Enter number of elements\n");
	scanf("%d",&iSize);

	p = (int *)malloc(iSize * sizeof(int));
	if(p == NULL)
	{
		printf("Unable to allocate memory\n");
		return -1;
	}
	printf("Enter %d elements\n",iSize);
	for(iCnt = 0;iCnt<iSize; iCnt++)
	{
		printf("Enter element : %d\n",iCnt+1);
		scanf("%d",&p[iCnt]);
	}
	iRet = Product(p, iSize);
	printf("Product is %d",iRet);
	free(p);
	return 0;
}

File: Assignment Programs/Assignment34/program5.java
import java.util.*;

class Number
{
	public int Product(int Arr[])
	{
		int iCnt = 0, iMult = 1, iFlag = 0;
		
		for(iCnt = Arr.length-1;iCnt>=0;iCnt--)
		{
			if((Arr[iCnt] % 2) != 0)
			{
				iMult = iMult * Arr[iCnt];
				iFlag = 1;
			}
		}
		if(iFlag == 0)
		{
			return 0;
		}
		else
		{
			return iMult;
		}
	}
}

class program5
{
	public static void main(String args[])
	{
		int iLength = 0, iCnt = 0, iRet = 0;
		boolean bRet = false;
		int Brr[];
		Scanner sobj = new Scanner(System.in);
		
		System.out.println("Enter the number of elements");
		iLength = sobj.nextInt();
		
		Brr = new int[iLength];
		
		System.out.println("Enter the elements");
		
		for(iCnt = 0;iCnt<Brr.length;iCnt++)
		{
			Brr[iCnt] = sobj.nextInt(); 
		}	
		
		Number nobj = new Number();
		
		iRet = nobj.Product(Brr);
		
		System.out.println("Product of odd elements from array is "+iRet);

	}
}